lib/pg_eventstore/client.rb

```diff
@@ -109,6 +109,15 @@ def read(stream, options: {}, middlewares: nil)
         call(stream, options: { max_count: config.max_count }.merge(options))
     end
 
+    # @see {#read} for available params
+    # @return [Enumerator] enumerator will yield PgEventstore::Event
+    def read_paginated(stream, options: {}, middlewares: nil)
+      cmd_class = stream.system? ? Commands::SystemStreamReadPaginated : Commands::RegularStreamReadPaginated
+      cmd_class.
+        new(Queries.new(streams: stream_queries, events: event_queries(middlewares(middlewares)))).
+        call(stream, options: { max_count: config.max_count }.merge(options))
```

Comment:
I wonder If you should not to do lazy load by default 
https://medium.com/rubycademy/the-enumerable-module-in-ruby-part-ii-41f69b36360
I think If someone just the code like this:
`PgEventstore.client.read_paginated(...).first(15)` it will reed anyway all events from the stream

Reply:
I am not quite sure what you mean. If you do this

```ruby
PgEventstore.client.read_paginated(PgEventstore::Stream.all_stream, options: { max_count: 2} ).first(2)
```

you will get two batches by 2 events

Reply:
What in case if in the stream we have 200k events and we do:  
`PgEventstore.client.read_paginated(PgEventstore::Stream.all_stream, options: { max_count: 2000} ).first(2)`

Reply:
it will load two batches by 2k events each

Reply:
sot `first(2)` is related to the batch or to the number of events I will get?

Reply:
Since the implementation yields batches - in your case `first(2)` will load 2 batches